From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 6 Aug 2016 11:27:01 +0200
Subject: [PATCH] {BUG in 4.8-rc1} drivers/tty/serial/8250/8250_fintek.c: fix
 redefinition of IRQ_MODE macro

    drivers/tty/serial/8250/8250_fintek.c:34:0: error: "IRQ_MODE" redefined [-Werror]
     #define IRQ_MODE 0x70

    In file included from arch/arm/include/asm/ptrace.h:13:0,
                     from arch/arm/include/asm/irqflags.h:6,
                     from include/linux/irqflags.h:15,
                     from arch/arm/include/asm/bitops.h:27,
                     from include/linux/bitops.h:36,
                     from include/linux/kernel.h:10,
                     from include/linux/list.h:8,
                     from include/linux/module.h:9,
                     from drivers/tty/serial/8250/8250_fintek.c:11:
    arch/arm/include/uapi/asm/ptrace.h:55:0: note: this is the location of the previous definition
     #define IRQ_MODE 0x00000012
---
 drivers/tty/serial/8250/8250_fintek.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/tty/serial/8250/8250_fintek.c b/drivers/tty/serial/8250/8250_fintek.c
index 737b4b3957b0..dad9f43b2496 100644
--- a/drivers/tty/serial/8250/8250_fintek.c
+++ b/drivers/tty/serial/8250/8250_fintek.c
@@ -31,6 +31,8 @@
 #define IO_ADDR2 0x60
 #define LDN 0x7
 
+/* BUG: IRQ_MODE is defined in arch/arm/include/uapi/asm/ptrace.h too */
+#undef IRQ_MODE
 #define IRQ_MODE	0x70
 #define IRQ_SHARE	BIT(4)
 #define IRQ_MODE_MASK	(BIT(6) | BIT(5))
-- 
